deps/corepack/dist/corepack.js

```diff
@@ -0,0 +1,14819 @@
+      throw er
+    try {
+      if (!opts.statSync(path).isDirectory())
+        throw er
+    } catch (_) {
+      throw er
+    }
+  }
+}
+
+module.exports = {mkdirpManual, mkdirpManualSync}
+
+
+/***/ }),
+
+/***/ "./.yarn/cache/mkdirp-npm-1.0.3-02fc91ac70-c5c46ae417.zip/node_modules/mkdirp/lib/mkdirp-native.js":
+/*!*********************************************************************************************************!*\
+  !*** ./.yarn/cache/mkdirp-npm-1.0.3-02fc91ac70-c5c46ae417.zip/node_modules/mkdirp/lib/mkdirp-native.js ***!
+  \*********************************************************************************************************/
+/***/ ((module, __unused_webpack_exports, __webpack_require__) => {
+
+const {dirname} = __webpack_require__(/*! path */ "path")
+const {findMade, findMadeSync} = __webpack_require__(/*! ./find-made.js */ "./.yarn/cache/mkdirp-npm-1.0.3-02fc91ac70-c5c46ae417.zip/node_modules/mkdirp/lib/find-made.js")
+const {mkdirpManual, mkdirpManualSync} = __webpack_require__(/*! ./mkdirp-manual.js */ "./.yarn/cache/mkdirp-npm-1.0.3-02fc91ac70-c5c46ae417.zip/node_modules/mkdirp/lib/mkdirp-manual.js")
+
+const mkdirpNative = (path, opts) => {
+  opts.recursive = true
+  const parent = dirname(path)
+  if (parent === path)
+    return opts.mkdirAsync(path, opts)
+
+  return findMade(opts, path).then(made =>
+    opts.mkdirAsync(path, opts).then(() => made)
+    .catch(er => {
+      if (er.code === 'ENOENT')
+        return mkdirpManual(path, opts)
+      else
+        throw er
+    }))
+}
+
+const mkdirpNativeSync = (path, opts) => {
+  opts.recursive = true
+  const parent = dirname(path)
+  if (parent === path)
+    return opts.mkdirSync(path, opts)
+
+  const made = findMadeSync(opts, path)
+  try {
+    opts.mkdirSync(path, opts)
+    return made
+  } catch (er) {
+    if (er.code === 'ENOENT')
+      return mkdirpManualSync(path, opts)
+    else
+      throw er
+  }
+}
+
+module.exports = {mkdirpNative, mkdirpNativeSync}
+
+
+/***/ }),
+
+/***/ "./.yarn/cache/mkdirp-npm-1.0.3-02fc91ac70-c5c46ae417.zip/node_modules/mkdirp/lib/opts-arg.js":
+/*!****************************************************************************************************!*\
+  !*** ./.yarn/cache/mkdirp-npm-1.0.3-02fc91ac70-c5c46ae417.zip/node_modules/mkdirp/lib/opts-arg.js ***!
+  \****************************************************************************************************/
+/***/ ((module, __unused_webpack_exports, __webpack_require__) => {
+
+const { promisify } = __webpack_require__(/*! util */ "util")
+const fs = __webpack_require__(/*! fs */ "fs")
+const optsArg = opts => {
+  if (!opts)
+    opts = { mode: 0o777 & (~process.umask()), fs }
+  else if (typeof opts === 'object')
+    opts = { mode: 0o777 & (~process.umask()), fs, ...opts }
+  else if (typeof opts === 'number')
+    opts = { mode: opts, fs }
+  else if (typeof opts === 'string')
+    opts = { mode: parseInt(opts, 8), fs }
+  else
+    throw 
```

Comment:
```console
$ sh /absolute/path/to/node/deps/corepack/shims/corepack --help
Error: Cannot find module 'corepack/package.json'
Require stack:
- /absolute/path/to/node/deps/corepack/dist/corepack.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:927:15)
    at Function.Module._load (node:internal/modules/cjs/loader:772:27)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:93:18)
    at main (/absolute/path/to/node/deps/corepack/dist/corepack.js:14787:46)
    at runMain (/absolute/path/to/node/deps/corepack/dist/corepack.js:14799:5)
    at /absolute/path/to/node/deps/corepack/dist/corepack.js:14811:5
    at /absolute/path/to/node/deps/corepack/dist/corepack.js:14813:3
    at Object.<anonymous> (/absolute/path/to/node/deps/corepack/dist/corepack.js:14818:12)
    at Module._compile (node:internal/modules/cjs/loader:1095:14)
```
If I replace this line with the following, it works as expected:
```suggestion
        const binaryVersion = '1.0.0';
```

Reply:
GitHub web UI is showing the wrong line, line 14787 currently contains:
```js
        const binaryVersion = eval(`require`)(`corepack/package.json`).version;
```

Reply:
Fixed upstream